resources/entity.rb

```diff
@@ -0,0 +1,53 @@
+#
+# Cookbook Name:: sensu-go
+# Resource:: agent
+#
+# Copyright:: 2018 Sensu, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+resource_name :sensu_entity
+
+action_class do
+  include SensuCookbook::Helpers
+end
+
+property :config_home, String, default: '/etc/sensu'
+
+property :subscriptions, Array
+property :organization, String, default: 'default'
+property :environment, String, default: 'default'
+property :class_, String
```

Comment:
I realize that this follows the upstream naming but I don't think it's a good one. `class` is a keyword in pretty much every programming language just with varying capitalization.

Reply:
In ruby, I thought the convention when explicitly referencing the underlying type system was to use the word `klass`. If that's the case, the `class_` seems appropriate. I'm curious what naming you'd suggest?

Reply:
I could also call it `entity_class` to be more specific. Would that work?

Reply:
I meant the upstream naming, yes the convention of `class_` given a terrible name like `class` I'd much prefer `entity_class` instead. Before 2.0 becomes GA maybe we should consider renaming that as this is an issue in pretty much every language.

Reply:
I went ahead and used `entity_class` and opened [2049](https://github.com/sensu/sensu-go/issues/2049) in sensu-go for some discussion on the naming.